src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -96,6 +99,90 @@ module.exports = function(scheduleUpdate : (fiber: Fiber, priorityLevel : Priori
     return true;
   }
 
+  function checkClassInstance(workInProgress: Fiber, inst: any) {
+    const type = workInProgress.type;
```

Comment:
Can we please not calculate these fields in production if we're not planning to print anything?

It's fine to extract this to `getFriendlyName(inst)` and only call this in `__DEV__` branch and in both-dev-and-prod `if ()` branch below when we're sure we'll print the error.